data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -148,6 +168,12 @@ private List<Predicate<T>> applyEqDeletes() {
 
       Schema deleteSchema = TypeUtil.select(requiredSchema, ids);
 
+      if (filePath.endsWith(".parquet") && parquetReader != null) {
```

Comment:
I think we can change the ctor parameter from `String` to `DataFile` and then here we can check file.format().

Reply:
yes, you are right, but dataFile is not passed into DeleteFilter as parameter, it was changed to filePath in #4381

Reply:
I see, how about adding another `ctor` to accept `DataFile` parameter?

Reply:
the change for constructor parameters is only for Trino supporting mor，Trino wrapped a dummy fileScanTask for data file currently, the author wants to remove fileScanTask implemented in Trino, and use the filePath parameter. If we change it to dataFile, compatibility is a problem.